### PR TITLE
Live comments

### LINKED ISSUE #2215: Live Comment Refreshing
**Describe the new addon**

This addon would get the comments on a studio, project, or profile every minute and show them without the user having to reload the page. It could work with the infinite scrolling addon, possibly, or be an option on that addon.

### PR DESCRIPTION
**Pull request by FunctionalMetatable**
Resolves #2215
Supersedes #3041

### Changes

On project and studio pages, new comments now load automatically (using redux)

### Reason for changes

Make it more like github

### Tests

Tested locally

cc @CubeyTheCube

### CODE REVIEW

**Comment by WorldLanguages** [comment]
@FunctionalMetatable What happens in edge cases? Such as writing a reply to a comment chain that later got deleted.

**Comment by FunctionalMetatable** [comment]
>  What happens in edge cases? Such as writing a reply to a comment chain that later got deleted.

Same as the Cubey's PR: 
> This doesn't remove deleted comments, because that might generate confusion

**Comment by CodeGuy92** [comment]
Let me test this locally.

**Comment by CodeGuy92** [comment]
Bruh, i tried turning this on and going to the TFH studio, and all of the comments start refreshing, but... backwards.
The new comments will start dissapearing, and after a few minutes, you will literally see comments from a year ago.

**Comment by FunctionalMetatable** [comment]
> Bruh, i tried turning this on and going to the TFH studio, and all of the comments start refreshing, but... backwards.
> The new comments will start dissapearing, and after a few minutes, you will literally see comments from a year ago.

hmm

**Comment by WorldLanguages** [comment]
Backlog, sorry. I'll probably get into this when I have the time, and I'm still not 100% sure whether this is a good idea.

**Comment by retronbv** [comment]
> if im correct, st wont like it

st doesn't even like scratch addons

**Comment by WorldLanguages** [comment]
> if im correct, st wont like it

Live comments or online statuses?

**Comment by Weredime** [comment]
any updates? this has been there for toooo long

File: addons/live-comments/addon.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "Live comment refreshing",
-  "description": "Refreshes comments to get a near-live view.",
+  "description": "Refreshes comments to get a near-live view (on studios and projects).",
```
Comment (mxmou) [inline addons/live-comments/addon.json:3]: I think this should say "refreshes comments on studios and projects to ...".

File: addons/live-comments/addon.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "Live comment refreshing",
+  "description": "Refreshes comments on studios and projects to create a near-live view.",
```
Comment (RedGuy12) [inline addons/live-comments/addon.json:3]: I think it'd be better to just say "Automatically refreshes website comments to create a near-live view." And specify that it doesn't work on user profiles in a `warning`

**Review by RedGuy12 — DISMISSED** [review]

File: addons/live-comments/addon.json (hunk 2 above)
Comment (RedGuy12) [reply addons/live-comments/addon.json:3]: i forgot those exist

**Comment by WorldLanguages** [comment]
What's the status of the requests I made on my review on https://github.com/ScratchAddons/ScratchAddons/pull/3064#issuecomment-888605040 ?

**Comment by Weredime** [comment]
> What's the status of the requests I made on my review on [#3064 (comment)](https://github.com/ScratchAddons/ScratchAddons/pull/3064#issuecomment-888605040) ?

TODO

**Comment by GrahamSH-LLK** [comment]
Is this still under development?

**Comment by apricotwtf** [comment]
Closing because #6171 has popped up, and I don't have the time to maintain this anymore (i am @FunctionalMetatable)
